### ISSUE TITLE
`getInputs` incorrectly interprets NSE for tidyr

### ISSUE THREAD

**Issue by ruaridhw** [duncantl/CodeDepends#32]
This may be related to #26.

`dplyr` works however `tidyr` doesn't. I haven't tried other `tidyverse` packages either though...

**Works**
```r
library(dplyr)
#> Warning: package 'dplyr' was built under R version 3.4.2
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
CodeDepends::getInputs(quote(
    iris %>%
      select(Petal.Width, Species)
))@inputs
#> Warning: replacing previous import 'graph::addNode' by 'XML::addNode' when
#> loading 'CodeDepends'
#> Warning: replacing previous import 'graph::plot' by 'graphics::plot' when
#> loading 'CodeDepends'
#> [1] "iris"
```
The only input is `iris` and `Petal.Width` & `Species` are ignored.

**Fails**
```r
library(dplyr)
#> Warning: package 'dplyr' was built under R version 3.4.2
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(tidyr)
#> Warning: package 'tidyr' was built under R version 3.4.3
CodeDepends::getInputs(quote(
    iris %>%
      select(Petal.Width, Species) %>%
      tidyr::gather(key = "var", value = "value", Petal.Width)
))@inputs
#> Warning: replacing previous import 'graph::addNode' by 'XML::addNode' when
#> loading 'CodeDepends'
#> Warning: replacing previous import 'graph::plot' by 'graphics::plot' when
#> loading 'CodeDepends'
#> [1] "Petal.Width" "iris"
```

`Petal.Width` is now an input when using `tidyr::gather`

**Comment by clarkfitzg** [comment]
CodeDepends has special logic to handle the nonstandard evaluation (NSE) in dplyr. Users are free to extend CodeDepends to handle NSE in any way they like. For @ruaridhw's example this will return `iris`:

```
getInputs(quote(
    iris %>%
      select(Petal.Width, Species) %>%
      gather(key = "var", value = "value", Petal.Width)
), collector = inputCollector(functionHandlers = list(gather = nseafterfirst))
)@inputs
```

**Comment by clarkfitzg** [comment]
I should mention that I don't know if `nseafterfirst` is correct here, because I don't know how `tidyr::gather` handles NSE. It's on the user to select the right function.

**Comment by gmbecker** [comment]
@ruaridhw generally @clarkfitzg is correct. The way to change how CodeDepends handles a particular function is by setting a function handler for that function, and my hope is that one of the existing function handlers will do most of the things people want without them having to actually roll their own. Particularly in the case of support NSE functions CodeDepends is not aware of.

In the case of gather, there was a bug in nsehandlerfactory that made the correct (nsehandlerfactory(3)) handler not work, but this has now been fixed in 0.5.5.6. I also added the `except` argument for the factory to support SE arguments that appear after a `...` which accepts NSE arguments. 

Finally, in this case, I've added `nsehandlerfactory(3, except = c("na.rm", "convert", "factor_key"))`  as the default handler for `gather`, and in general you can do the same thing next time you need custom NSE support. Do also inform us when you run into this, and we will consider adding it to the defaults if there is  not a good reason not to.

**Comment by ruaridhw** [comment]
Thank you both for the clear explanation.

I came to CodeDepends from ropensci/drake#320 and ropensci/drake#268 and assumed the second case would cause problems. As it turns out, `drake` seems to take care of this itself.
